docs(tutorial): add deprecation notice to v1.4.x tutorial and point to the correct branch

gkalpak opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Docs update.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] ~~Tests for the changes have been added (for bug fixes / features)~~
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:
- Add a deprecation notice on the 'index' page.
- Hide the "Live Demo" buttons (since we don't have a live demo).
- Update the GitHub diff links to point to the ~~`pre-v1.5.0-snapshot`~~ `1.4-snapshot` angular-phonecat branch.
- Modify all git commangs to use the appropriate branch and tags.

Related to #14416.

gkalpak committed:
docs(tutorial): add deprecation notice to v1.4.x tutorial and point to the correct branch

- Add a deprecation notice on the 'index' page.
- Hide the "Live Demo" buttons (since we don't have a live demo).
- Update the GitHub diff links to point to the `pre-v1.5.0-snapshot` angular-phonecat branch.
- Modify all git commangs to use the appropriate branch and tags.

Related to #14416.

gkalpak commented:
The only thing missing (except for merging this PR) is to create the `old-step-*` tags on angular-phonecat.

(We might need better names for `pre-v1.5.0-snapshot` and `old-step-*`.)

gkalpak commented:
@petebacondarwin, I have assigned you, since you are the official tutorial reviewer this month :stuck_out_tongue:

petebacondarwin commented on docs/app/src/tutorials.js:
```diff
@@ -11,16 +11,16 @@ angular.module('tutorials', [])
-      scope.diffLo = seq ? (seq - 1): '0~1';
-      scope.diffHi = seq;
+      scope.diffLo = 13 - seq;
+      scope.diffHi = 12 - seq;
```
why these changes?

gkalpak replied:
Hm...I made them before I created the new tags. Now that we have the tags, I should just change `compare/step-` to `compare/old-step-` :smiley:

petebacondarwin commented on docs/app/src/tutorials.js:
```diff
@@ -11,16 +11,16 @@ angular.module('tutorials', [])
       '<a ng-href="tutorial/{{prev}}"><li class="btn btn-primary"><i class="glyphicon glyphicon-step-backward"></i> Previous</li></a>\n' +
-      '<a ng-href="http://angular.github.io/angular-phonecat/step-{{seq}}/app"><li class="btn btn-primary"><i class="glyphicon glyphicon-play"></i> Live Demo</li></a>\n' +
-      '<a ng-href="https://github.com/angular/angular-phonecat/compare/step-{{diffLo}}...step-{{diffHi}}"><li class="btn btn-primary"><i class="glyphicon glyphicon-search"></i> Code Diff</li></a>\n' +
+      // '<a ng-href="http://angular.github.io/angular-phonecat/step-{{seq}}/app"><li class="btn btn-primary"><i class="glyphicon glyphicon-play"></i> Live Demo</li></a>\n' +
```
leaving this commented for the future?

gkalpak replied:
For historical reasons :stuck_out_tongue: I will remove it.
Are we OK with not having a live demo for the old tutorial ?

petebacondarwin replied:
Yes

petebacondarwin commented on docs/content/tutorial/index.ngdoc:
```diff
@@ -5,6 +5,22 @@
+    </ul>
+  </p>
+</div>
+
```
LG

petebacondarwin commented on docs/content/tutorial/index.ngdoc:
````diff
@@ -73,7 +89,7 @@ Clone the [angular-phonecat repository][angular-phonecat] located at GitHub by r
 
 ```
-git clone --depth=14 https://github.com/angular/angular-phonecat.git
+git clone --depth=14 --branch=pre-v1.5.0-snapshot https://github.com/angular/angular-phonecat.git
````
LG

petebacondarwin commented on docs/content/tutorial/index.ngdoc:
```diff
@@ -82,6 +98,11 @@ This command creates the `angular-phonecat` directory in your current directory.
+  The `--branch=pre-v1.5.0-snapshot` option tells Git to pull down the `pre-v1.5.0-snapshot` branch.
+  The code for this older version of the tutorial is on that branch.
+</div>
+
```
LG

petebacondarwin commented on docs/content/tutorial/step_00.ngdoc:
````diff
@@ -16,7 +16,7 @@ dependencies, as described in {@link index#get-started Get Started}.
 
 ```
-git checkout -f step-0
+git checkout -f old-step-0
````
LG

gkalpak committed:
fixup 1

gkalpak commented:
@petebacondarwin, I made the changes. PTAL

gkalpak committed:
fixup 2

- Rename branch: `pre-v1.5.0-snapshot` --> `1.4-snapshot`
- Rename steps:  `old-step-*`          --> `1.4-step-*`

gkalpak commented:
Renamed branch to `1.4-snapshot` and tags to `1.4-step-*` (as discussed off-github).
Merging...

gkalpak commented:
Closed via 6234cda.
